src/arch/riscv/kernel/vspace.c

```diff
@@ -180,8 +180,8 @@ BOOT_CODE void map_it_pt_cap(cap_t vspace_cap, cap_t pt_cap)
     *targetSlot = pte_new(
```

Comment:
It might be a pain for the proofs, but having a `pte_new_leaf()` and `pte_new_table()` might be useful to make the code more clear.

Reply:
We could wrap `pte_new`, I don't think it would be too catastrophic, but I'd first like to explore if we can rescue the uniform encoding. If not this is going to be months of work, because it kills one of the fundamental formalisation assumptions.

Reply:
Would be something like https://github.com/axel-h/seL4/commit/060b0cdee851a4aecc49b5f7d9f9a6ac31b67428 then that I had in mind